x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/filter_agg/config.ts

```diff
@@ -199,6 +211,18 @@ export function getFilterAggTypeConfig(
         getEsAggConfig() {
           return this.filterAggConfig !== undefined ? JSON.parse(this.filterAggConfig!) : {};
         },
+        isValid() {
+          if (typeof this.filterAggConfig === 'string') {
+            try {
+              return !!JSON.parse(this.filterAggConfig);
+            } catch (e) {
+              // eslint-disable-next-line no-console
+              console.error(`Invalid JSON for aggregation definition.\n${e}`);
+              return false;
+            }
+          }
+          return true;
+        },
```

Comment:
Would be great to avoid code repetition. I reckon it's worth creating a reusable `jsonValidator`, similar to what we have with the [`numberValidator`](https://github.com/elastic/kibana/blob/6b5101014132678f6aabe936a9a96fdea2f6dc85/x-pack/packages/ml/agg_utils/src/validate_number.ts#L19).

Reply:
Updated here https://github.com/elastic/kibana/pull/137773/commits/6cde9458bb3ef80020abbb7b3024bfa929a165bc